File: backend/src/main/java/ecrow/backend/business/abstracts/TownService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.entities.concretes.Town;

import java.util.List;

public interface TownService {
    DataResult<List<Town>> getAll();
    DataResult<Town> getById(Integer id);
    DataResult<List<Town>> getByFkCityId(Integer cityId);
}

File: frontend/src/services/TownService.js
import axios from "axios";

export default class TownService{
    getAll() {
        return axios.get("http://localhost:8080/api/town/getAll")
    }

    getById(id) {
        return axios.get(`http://localhost:8080/api/town/getById?id=${id}`)
    }

    getByCityId(cityId){
        return axios.get(`http://localhost:8080/api/town/getByFkCityId?cityId=${cityId}`)
    }
}